projects/ion/src/lib/popover/popover.directive.ts

```diff
@@ -154,21 +160,27 @@ export class IonPopoverDirective implements OnDestroy {
 
   @HostListener('click') onClick(): void {
     const hostElement = this.viewRef.element.nativeElement as HTMLElement;
-    const position = hostElement.getBoundingClientRect();
-
-    if (this.elementIsEnabled(hostElement)) {
-      this.open(position);
+    if (this.elementIsEnabled(hostElement) && this.isComponentRefNull()) {
+      this.open();
     }
   }
 
+  @HostListener('window:scroll') onScroll(): void {
```

Comment:
To help to solve the problem where popover doesn't close correctly if is not window that has the scrollbar, we can add some more events, like `document:scroll`, `body:scroll` and, primarly, `window:wheel`, to listen or detect any scrolls on the page.

Reply:
OK, I'll add these events